test/controllers/calendar_controller_test.exs

```diff
@@ -0,0 +1,432 @@
+      student_conn = login_as( test_data.student_with_no_class )
+
+      params = %{date: "2017-01-01", my_classes: "true"}
+      conn = get student_conn, calendar_path(student_conn, :index), params
+      assert json_response(conn, 200) == @empty_result
+    end
+
+    @tag user_role: :student
+    test "all classes when my_classes parameter is not present", %{conn: conn} do
+      create_test_data()
+
+      expected_result =
+        %{"classes" => [%{"classroom" => "r101",
+                          "date" => "2017-01-01",
+                          "finishes_at" => "12:00:00",
+                          "starting_at" => "10:00:00",
+                          "course_name" => "english",
+                          "teachers" => [%{"family_name" => "tf1",
+                                           "nickname" => "tn1",
+                                           "name" => "t1"}],
+                          "term_name" => "term1"}]}
+
+      params = %{date: "2017-01-01"}
+      conn = get conn, calendar_path(conn, :index), params
+      assert json_response(conn, 200) == expected_result
+    end
+
+    @tag user_role: :student
+    test "all classes for student when my_classes parameter is false", %{conn: conn} do
+      create_test_data()
+
+      expected_result =
+        %{"classes" => [%{"classroom" => "r101",
+                          "date" => "2017-01-01",
+                          "finishes_at" => "12:00:00",
+                          "starting_at" => "10:00:00",
+                          "course_name" => "english",
+                          "teachers" => [%{"family_name" => "tf1",
+                                           "nickname" => "tn1",
+                                           "name" => "t1"}],
+                          "term_name" => "term1"}]}
+
+      params = %{date: "2017-01-01", my_classes: "false"}
+      conn = get conn, calendar_path(conn, :index), params
+      assert json_response(conn, 200) == expected_result
+    end
+
+    @tag user_role: :student
+    test "attending classes when my_classes parameter is true", %{conn: _conn} do
+      test_data = create_test_data()
+      student = List.last(test_data.students)
+      studen_conn = login_as(student)
+
+      expected_result =
+        %{"classes" => [%{"classroom" => "r105",
+                          "date" => "2017-01-05",
+                          "finishes_at" => "12:00:00",
+                          "starting_at" => "10:00:00",
+                          "course_name" => "english",
+                          "teachers" => [%{"family_name" => "tf1",
+                                           "nickname" => "tn1",
+                                           "name" => "t1"},
+                                         %{"family_name" => "tf2",
+                                            "nickname" => "tn2",
+                                            "name" => "t2"}],
+                          "term_name" => "term1"}]}
```

Comment:
Can you extract this expected result? It's probably occupying half of the file's lines and as I skimmed through, looks mostly identical in all tests.